query/search_intersections.js

```diff
@@ -0,0 +1,227 @@
+    });
+    logStr += '[param:boundary_country] ';
+  }
+
+  // run the address parser
+  if( clean.parsed_text ){
+    textParser( clean.parsed_text, vs );
+  }
+
+  var q = getQuery(vs, clean);
+
+  if (q !== undefined) {
+    logger.info(logStr);
+  }
+  else {
+    logger.info('[parser:intersections] query type not supported');
+  }
+
+  return q;
+}
+
+function getQuery(vs, clean) {
+  logger.info(`[query:search] [search_input_type:${determineQueryType(vs)}]`);
+
+  if (hasIntersection(vs)) {
+    return generateIntersectionQuery(vs,clean);
+  }
+
+  // returning undefined is a signal to a later step that the addressit-parsed
+  // query should be queried for
+  return undefined;
+
+}
+
+function determineQueryType(vs) {
+  if (vs.isset('input:housenumber') && vs.isset('input:street')) {
+    return 'address';
+  }
+  else if (vs.isset('input:street')) {
+    return 'street';
+  }
+  else if (vs.isset('input:street1') || vs.isset('input:street2')) {
+    return 'intersection';
+  }
+  else if (vs.isset('input:query')) {
+    return 'venue';
+  }
+  else if (['neighbourhood', 'borough', 'postcode', 'county', 'region','country'].some(
+    layer => vs.isset(`input:${layer}`)
+  )) {
+    return 'admin';
+  }
+  return 'other';
+}
+
+function hasIntersection(vs) {
+  return vs.isset('input:street1') && vs.isset('input:street2');
+}
+
+function shouldSort(vs) {
+  if (vs.isset('focus:point:lat') && vs.isset('focus:point:lon')) {
+    return true;
+  }
+
+  return false;
+}
+
+function generateIntersectionQuery(vs,clean) {
+  let sort = {};
+  if (shouldSort(vs)) {
+    sort = {
+              '_geo_distance': {
+                  'order': 'asc',
+                  'distance_type': 'plane',
+                  'center_point': {
+                      'lat': clean['focus.point.lat'],
+                      'lon': clean['focus.point.lon']
+                  }
+              }
+    };
+  }
+
+  return {
+      type: 'fallback',
+      body: {
+          'size': 2,
+          'query': {
+              'or': [
+                  {
+                      'bool': {
+                          'must': [
+                              { 'match': { 'layer': 'intersection'} },
+                              { 'match': { 'address_parts.street1': clean.parsed_text.street1} },
+                              { 'match': { 'address_parts.street2': clean.parsed_text.street2} }
+                          ]
+                      }
+                  },
+                  {
+                      'bool': {
+                          'must': [
+                              { 'match': { 'layer': 'intersection'} },
+                              { 'match': { 'address_parts.street1': clean.parsed_text.street2} },
+                              { 'match': { 'address_parts.street2': clean.parsed_text.street1} }
+                          ]
+                      }
+                  }
+              ]
+          },
+          'sort': [sort]
+      }
```

Comment:
We probably want to add additional filters to this query so that it matches the rest of our queries. Layers, sources, etc. @trescube will know more and also can advise if we want to handle any sort of fallback (to street, city, etc) logic here.

Reply:
@orangejulius Right. I have some suggestions, too. Would love to talk.